### DISCUSSION TITLE
v5.2 add, modify and rename utilities the right way?

### DISCUSSION THREAD

**Discussion by aboutjquery** [twbs/bootstrap#36437] [General] [answered]
I am using gulp to compile scss.

Add, Modify, Rename utilities are only working between @import "bootstrap/scss/utilities"; and @import "bootstrap/scss/utilities/api";
Just like
```
@import "bootstrap/scss/functions";
@import "bootstrap/scss/variables";
@import "bootstrap/scss/maps";
@import "bootstrap/scss/mixins";
@import "bootstrap/scss/utilities";
$utilities: map-merge(
  $utilities,
  (
    "width": map-merge(
      map-get($utilities, "width"),
      (
        values: map-merge(
          map-get(map-get($utilities, "width"), "values"),
          (10: 10%),
        ),
      ),
    ),
  )
);

...other components scss
@import "bootstrap/scss/utilities/api";
```

But the demonstration of the official document is to insert the code of Add, Modify, Rename utilities after the following

```
@import "bootstrap/scss/functions";
@import "bootstrap/scss/variables";
@import "bootstrap/scss/maps";
@import "bootstrap/scss/mixins";
@import "bootstrap/scss/utilities";
...other components scss
@import "bootstrap/scss/utilities/api";
$utilities: map-merge(
  $utilities,
  (
    "width": map-merge(
      map-get($utilities, "width"),
      (
        values: map-merge(
          map-get(map-get($utilities, "width"), "values"),
          (10: 10%),
        ),
      ),
    ),
  )
);
```

Am I wrong or is there something wrong with the official documentation?

thank you

**Comment by julien-deramond** [comment, marked as answer]
You're not wrong, there's something wrong with the documentation:

> the customization of the $utilities map needs to come before the api import, but after the rest of the imports. (source: https://github.com/twbs/bootstrap/issues/36436#issuecomment-1138131325)

The documentation will be fixed in the 5.2.0.
